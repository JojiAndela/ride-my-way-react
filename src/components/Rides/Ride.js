import React from 'react'
import PropTypes from 'prop-types';
import classes from './Rides.scss';

const Ride = (props) => (
    <div className={classes.Ride} onClick={props.clicked}>
      <h2>{props.title}</h2>
    <p> From <strong>{props.location}</strong> to <strong>{props.destination}</strong></p>
      <p> <b>{props.slot}</b>  Seat Available</p>

      <div className={classes.driver}>
      <span><p><strong>By:</strong> {props.driver}</p>
    <p><strong>Departure date:</strong> {props.date}</p>
  <p><strong>Departure time:</strong> {props.time}</p>
  </span>
      </div>
    </div>
  );

  Ride.propTypes ={
  title: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  destination: PropTypes.string.isRequired,
  slot: PropTypes.number.isRequired,
  driver: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  clicked: PropTypes.func.isRequired
};

export default Ride;
