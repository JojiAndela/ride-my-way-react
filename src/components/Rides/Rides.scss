.Rides{
  display: inline-grid;
  grid-template-columns: auto auto auto;
  grid-gap: 20px;
  width: 100%;
}

.Ride {
  padding: 10px;
  border: 1px #eee solid;
  border-radius: 5px;
  box-shadow: 2px 2px 8px #eee;
  margin: 10px;
  cursor: pointer;
  &:hover{
    background: #eee;
  }
  h2 {
    text-align: left;
    color: #444;
  }

  .driver {
    border: 1px #ccc solid;
    padding: 0 10px;
    border-radius: 5px;
    box-shadow: 2px 2px 2px #eee;
  }
}

@media screen and (max-width: 700px){
  .Rides{
    grid-template-columns: auto;
  }
}

@media screen and (max-width: 1090px) and (min-width: 690px){
  .Rides{
    grid-template-columns: auto auto;
  }
}

@media screen and (min-width: 1450px){
  .Rides{
    grid-template-columns: auto auto auto auto;
  }
}
