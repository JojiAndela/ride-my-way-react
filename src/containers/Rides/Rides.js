import React, { Component } from 'react';


class Rides extends Component {

  render(){
    return (
      <div> Rides </div>
    );
  }
}

export default Rides;
