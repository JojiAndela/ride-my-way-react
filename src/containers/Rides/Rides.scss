.Rides {
  padding: 20px;
  display: flex;
  // flex-direction: column;
  justify-content: center;
  align-content:center;

}
  h2 {
    text-align: center;
  }