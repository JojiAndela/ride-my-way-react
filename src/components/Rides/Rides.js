import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import timeConvert from 'Utility/timeConvert';
import classes from './Rides.scss';
import Ride from './Ride';

const Rides = (props) => {
  const { rides } = props;
  const mapped = rides.map(ride =>{
    return <Ride
        key={ride.id}
        id={ride.id}
        title={ride.name}
        destination={ride.destination}
        location={ride.location}
        slot={ride.slot}
        driver={ride.driver}
        date={ride.created_at.split("T")[0].replace(/-/g, " ")}
        time={timeConvert(ride.time)}
        clicked={() => props.history.push(`/dashboard/rides/${ride.id}`)}
      />
    });
  return (
    <div className={classes.Rides}>
      {mapped}
    </div>
  )
}

Rides.propTypes = {
  history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
  rides: PropTypes.array.isRequired
}
const mapStateToProps = state => ({
  rides: state.rides.rides,
});


export default connect(mapStateToProps)(withRouter(Rides));
