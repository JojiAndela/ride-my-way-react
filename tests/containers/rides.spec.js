import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

// Component to be tested
import Rides from '../../src/containers/Rides/Rides';

const initialState ={
  rides: {
    loading: false,
    rides: [],
    singleRide: null,
    requests: [],
    myRequest: {}
  }
}


const middleware = [thunk];
const mockStore = configureStore(middleware);
let wrapper;
let store;
describe('Rides', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Rides
          />
        </BrowserRouter>
      </Provider>
    );
  });
  it('should render the rides page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
