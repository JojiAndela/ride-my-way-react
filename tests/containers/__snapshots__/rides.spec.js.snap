// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rides should render the rides page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Rides)>
        <Rides
          getRides={[Function]}
          loading={false}
        >
          <Aux>
            <h2>
              All Rides
            </h2>
            <div
              className="Rides"
            >
              <Connect(withRouter(Rides))>
                <withRouter(Rides)
                  dispatch={[Function]}
                  rides={Array []}
                >
                  <Route>
                    <Rides
                      dispatch={[Function]}
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      rides={Array []}
                    >
                      <div
                        className="Rides"
                      />
                    </Rides>
                  </Route>
                </withRouter(Rides)>
              </Connect(withRouter(Rides))>
            </div>
          </Aux>
        </Rides>
      </Connect(Rides)>
    </Router>
  </BrowserRouter>
</Provider>
`;
